pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -96,34 +170,40 @@ func NewPortTable(start, end int) (*PortTable, error) {
 func (pt *PortTable) CleanupAllEntries() {
 	pt.tableLock.Lock()
 	defer pt.tableLock.Unlock()
-	pt.NodePortTable = make(map[string]*NodePortData)
-	pt.PodEndpointTable = make(map[string]*NodePortData)
+	pt.releaseDataFromPortTableCache()
 }
 
-func (pt *PortTable) GetDataForPodIP(ip string) []NodePortData {
+func (pt *PortTable) GetDataForPodIP(ip string) []*NodePortData {
 	pt.tableLock.RLock()
 	defer pt.tableLock.RUnlock()
 	return pt.getDataForPodIP(ip)
 }
 
-func (pt *PortTable) getDataForPodIP(ip string) []NodePortData {
-	var allData []NodePortData
-	for i := range pt.NodePortTable {
-		if pt.NodePortTable[i].PodIP == ip {
-			allData = append(allData, *pt.NodePortTable[i])
-		}
+func (pt *PortTable) getDataForPodIP(ip string) []*NodePortData {
+	allData, exist := pt.getPortTableCacheFromPodIPIndex(ip)
+	if exist == false {
+		return nil
 	}
 	return allData
 }
 
-func (pt *PortTable) getEntryByPodIPPort(ip string, port int) *NodePortData {
-	return pt.PodEndpointTable[podIPPortFormat(ip, port)]
+// podIPPortProtoFormat formats the ip, port and protocol to string ip:port:protocol.
+func podIPPortProtoFormat(ip string, port int, protocol string) string {
+	return fmt.Sprintf("%s:%d:%s", ip, port, protocol)
+}
+
+func (pt *PortTable) getEntryByPodIPPortProto(ip string, port int, protocol string) *NodePortData {
```

Comment:
> Have you made this change, I don't see it?
@antoninbas I moved getEntryByPodIPPortProto here.

Reply:
I was talking about `GetEntry`

Reply:
My bad. Move it to port_table.go, thanks.